data-prepper-plugins/ruby-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/ruby/RubyProcessor.java

```diff
@@ -57,20 +62,52 @@ public Collection<Record<Event>> doExecute(final Collection<Record<Event>> recor
 
     private void injectAndProcessEvents(List<Event> events) {
         container.put("events", events);
-        String script = "events.each { |event| \n"
+        script = "events.each { |event| \n"
                 + this.codeToInject +
                 "}";
 
         try {
             container.runScriptlet(script);
         } catch (EvalFailedException exception) {
             LOG.error("Exception processing Ruby code", exception);
-            if (!config.isIgnoreException()) {
-                throw new RuntimeException(); // todo: will this break the pipeline, as intended?
+
+            if (exceptionIsNameError(exception) && nameErrorIsEvent(exception.getMessage()))
+            {
+                LOG.debug("NameError on JacksonEvent caught, trying to recover via overloaded Event");
+
+                container.put("events", events);
+                container.runScriptlet(cleanEventAPICalls(script));
+            }
+            else {
+                if (!config.isIgnoreException()) {
+                    throw new RuntimeException(); // todo: should we tag instead of breaking pipeline?
+                }
             }
         }
     }
 
+    private String cleanEventAPICalls(String script) {
```

Comment:
I like the idea of this method, but I think the implementation can be improved. Does the Ruby code the customer supplies even need to specify `<X>.class`? Could it instead be `event.get("key")` converts to `event.get("key", java.lang.Object.java_class)`?